cmd/kubeadm/app/preflight/checks_darwin.go

```diff
@@ -25,3 +25,9 @@ package preflight
 func (idsc IsDockerSystemdCheck) Check() (warnings, errorList []error) {
 	return nil, nil
 }
+
+// Check determines if IPVS proxier can be used or not
+// No-op for for Darwin (MacOS).
+func (ipvspc IPVSProxierCheck) Check() (warnings, errors []error) {
```

Comment:
Unsupported check should not return an error, I believe.
You can look at other unsupported check in the same file: IsDockerSystemdCheck. Systemd doesn't exist on MacOS(the same as IPVSProxier), so check is just returns nil, nil.

Reply:
@bart0sh is correct, unsupported checks do not return an error, if they did then the check would fail for Mac or Windows and kubeadm would be unable to proceed without the user adding command line options to explicitly ignore the check.

Reply:
I'm still a bit confused on why the Check should not return even a warning on a platform that did not support it. Perhaps the warning/error is exposed elsewhere since I don't believe IPVS will work if we try to leverage kubeadm to install this on MacOS.

Reply:
@cmluciano this is a standard pattern in a variety of programming languages, golang including.
you have separate OS files where if a certain functionality does not work, it is a no-op and does not throw any errors.